<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1401</width>
    <height>1002</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QFrame" name="frame_buttons">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QPushButton" name="btnCreateRoot">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Create Root CA</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnCreateSub">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Create Sub CA</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnCreateEndEntiy">
         <property name="text">
          <string>Create End Entity</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnManage">
         <property name="text">
          <string>Manage CA</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="btnDelete">
         <property name="text">
          <string>Delete</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame_forms">
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QTabWidget" name="tabWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="tab">
          <attribute name="title">
           <string>CA Tree</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_7">
           <item row="0" column="0">
            <widget class="QTreeView" name="treeView">
             <property name="sizeAdjustPolicy">
              <enum>QAbstractScrollArea::AdjustIgnored</enum>
             </property>
             <property name="showDropIndicator" stdset="0">
              <bool>true</bool>
             </property>
             <property name="alternatingRowColors">
              <bool>true</bool>
             </property>
             <property name="iconSize">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
             <property name="textElideMode">
              <enum>Qt::ElideLeft</enum>
             </property>
             <property name="autoExpandDelay">
              <number>5</number>
             </property>
             <property name="sortingEnabled">
              <bool>true</bool>
             </property>
             <attribute name="headerCascadingSectionResizes">
              <bool>true</bool>
             </attribute>
             <attribute name="headerDefaultSectionSize">
              <number>100</number>
             </attribute>
             <attribute name="headerMinimumSectionSize">
              <number>100</number>
             </attribute>
             <attribute name="headerStretchLastSection">
              <bool>true</bool>
             </attribute>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="tab_2">
          <attribute name="title">
           <string>End Entities</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_8">
           <item row="0" column="0">
            <widget class="QTableView" name="tblEndEntities"/>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
       <item>
        <widget class="QFrame" name="frame">
         <property name="frameShape">
          <enum>QFrame::StyledPanel</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Raised</enum>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_3">
          <item>
           <layout class="QFormLayout" name="layoutRight">
            <property name="labelAlignment">
             <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
            </property>
            <item row="0" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>Name</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="leName"/>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Description</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="leDescription"/>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>UID</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLineEdit" name="leUID">
              <property name="acceptDrops">
               <bool>false</bool>
              </property>
              <property name="readOnly">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Common Name CN=</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QLineEdit" name="leCommonName"/>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>Organisational Unit OU=</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QLineEdit" name="leOrgUnit"/>
            </item>
            <item row="5" column="0">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Organization O=</string>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QLineEdit" name="leOrganization"/>
            </item>
            <item row="6" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Locality L=</string>
              </property>
             </widget>
            </item>
            <item row="6" column="1">
             <widget class="QLineEdit" name="leLocality"/>
            </item>
            <item row="7" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>State or Province SP=</string>
              </property>
             </widget>
            </item>
            <item row="7" column="1">
             <widget class="QLineEdit" name="leStateOrProvince"/>
            </item>
            <item row="8" column="0">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Country C=</string>
              </property>
             </widget>
            </item>
            <item row="8" column="1">
             <widget class="QLineEdit" name="leCountry"/>
            </item>
            <item row="9" column="0">
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>e-Mail</string>
              </property>
             </widget>
            </item>
            <item row="9" column="1">
             <widget class="QLineEdit" name="leEmail"/>
            </item>
            <item row="10" column="0">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>Domain</string>
              </property>
             </widget>
            </item>
            <item row="10" column="1">
             <widget class="QLineEdit" name="leDomain"/>
            </item>
            <item row="11" column="0">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Subject Alternative Names</string>
              </property>
             </widget>
            </item>
            <item row="11" column="1">
             <widget class="QFrame" name="frame_3">
              <property name="frameShape">
               <enum>QFrame::StyledPanel</enum>
              </property>
              <property name="frameShadow">
               <enum>QFrame::Raised</enum>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <property name="leftMargin">
                <number>0</number>
               </property>
               <property name="topMargin">
                <number>0</number>
               </property>
               <property name="rightMargin">
                <number>0</number>
               </property>
               <property name="bottomMargin">
                <number>0</number>
               </property>
               <item>
                <widget class="QListView" name="lstSubjectAltName">
                 <property name="maximumSize">
                  <size>
                   <width>16777215</width>
                   <height>100</height>
                  </size>
                 </property>
                </widget>
               </item>
               <item>
                <layout class="QVBoxLayout" name="verticalLayout_2">
                 <property name="leftMargin">
                  <number>0</number>
                 </property>
                 <property name="topMargin">
                  <number>10</number>
                 </property>
                 <property name="rightMargin">
                  <number>0</number>
                 </property>
                 <property name="bottomMargin">
                  <number>10</number>
                 </property>
                 <item>
                  <widget class="QToolButton" name="btnAltNameAdd">
                   <property name="text">
                    <string>+</string>
                   </property>
                   <property name="icon">
                    <iconset>
                     <normaloff>../../Downloads/Plus-48.png</normaloff>../../Downloads/Plus-48.png</iconset>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <widget class="QToolButton" name="btnAltNameDel">
                   <property name="text">
                    <string>-</string>
                   </property>
                   <property name="icon">
                    <iconset>
                     <normaloff>../../Downloads/Minus-48.png</normaloff>../../Downloads/Minus-48.png</iconset>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item row="12" column="0">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>Start Date</string>
              </property>
             </widget>
            </item>
            <item row="12" column="1">
             <widget class="QDateTimeEdit" name="dtStart">
              <property name="calendarPopup">
               <bool>true</bool>
              </property>
             </widget>
            </item>
            <item row="13" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>End Date</string>
              </property>
             </widget>
            </item>
            <item row="13" column="1">
             <widget class="QDateTimeEdit" name="dtEnd">
              <property name="calendarPopup">
               <bool>true</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QFrame" name="frame_2">
            <property name="frameShape">
             <enum>QFrame::StyledPanel</enum>
            </property>
            <property name="frameShadow">
             <enum>QFrame::Raised</enum>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <item>
              <widget class="QGroupBox" name="rfcUsage">
               <property name="title">
                <string>RFC5759 Suite B Key Usage Compliance</string>
               </property>
               <layout class="QGridLayout" name="gridLayout">
                <item row="0" column="1">
                 <widget class="QRadioButton" name="rbKeyEstablishment">
                  <property name="text">
                   <string>Key Establishment</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <widget class="QRadioButton" name="rbNonCompliant">
                  <property name="text">
                   <string>Non Compliant</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QRadioButton" name="rbDigitalSig">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="text">
                   <string>Digital Signature</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox_3">
               <property name="title">
                <string>Key Length</string>
               </property>
               <layout class="QGridLayout" name="gridLayout_4">
                <item row="0" column="0">
                 <widget class="QComboBox" name="KeyLength">
                  <item>
                   <property name="text">
                    <string>256</string>
                   </property>
                  </item>
                  <item>
                   <property name="text">
                    <string>384</string>
                   </property>
                  </item>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox_2">
               <property name="title">
                <string>Basic Constraints</string>
               </property>
               <layout class="QGridLayout" name="gridLayout_3">
                <item row="0" column="1">
                 <widget class="QLabel" name="label_17">
                  <property name="text">
                   <string>PathLength=</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QCheckBox" name="CA">
                  <property name="enabled">
                   <bool>true</bool>
                  </property>
                  <property name="text">
                   <string>CA</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <widget class="QLineEdit" name="lePathLength"/>
                </item>
               </layout>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="groupBox">
               <property name="title">
                <string>Key Usage</string>
               </property>
               <layout class="QGridLayout" name="gridLayout_2">
                <item row="2" column="0">
                 <widget class="QCheckBox" name="cRLSign">
                  <property name="text">
                   <string>cRLSign</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="1">
                 <widget class="QCheckBox" name="contentCommitment">
                  <property name="text">
                   <string>contentCommitment</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QCheckBox" name="keyAgreement">
                  <property name="text">
                   <string>keyAgreement</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QCheckBox" name="encipherOnly">
                  <property name="text">
                   <string>encipherOnly</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QCheckBox" name="dataEncipherment">
                  <property name="text">
                   <string>dataEncipherment</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="0">
                 <widget class="QCheckBox" name="digitalSignature">
                  <property name="text">
                   <string>digitalSignature</string>
                  </property>
                 </widget>
                </item>
                <item row="0" column="2">
                 <widget class="QCheckBox" name="keyEncipherment">
                  <property name="text">
                   <string>keyEncipherment</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="2">
                 <widget class="QCheckBox" name="keyCertSign">
                  <property name="text">
                   <string>keyCertSign</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="2">
                 <widget class="QCheckBox" name="decipherOnly">
                  <property name="text">
                   <string>decipherOnly</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_4">
            <property name="title">
             <string/>
            </property>
            <layout class="QGridLayout" name="gridLayout_5">
             <item row="0" column="1">
              <widget class="QPushButton" name="btnGenCSR">
               <property name="text">
                <string>Generate CSR</string>
               </property>
              </widget>
             </item>
             <item row="0" column="0">
              <widget class="QPushButton" name="btnGenKey">
               <property name="text">
                <string>Generate Key</string>
               </property>
              </widget>
             </item>
             <item row="0" column="2">
              <widget class="QPushButton" name="btnGenCert">
               <property name="text">
                <string>Sign CSR / Gen Cert</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="gridLayout_6">
         <property name="leftMargin">
          <number>10</number>
         </property>
         <property name="topMargin">
          <number>10</number>
         </property>
         <property name="rightMargin">
          <number>10</number>
         </property>
         <property name="bottomMargin">
          <number>10</number>
         </property>
         <item row="1" column="0">
          <widget class="QPlainTextEdit" name="PrivateKey">
           <property name="font">
            <font>
             <family>Consolas</family>
            </font>
           </property>
           <property name="lineWrapMode">
            <enum>QPlainTextEdit::NoWrap</enum>
           </property>
           <property name="readOnly">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QPlainTextEdit" name="Certificate">
           <property name="font">
            <font>
             <family>Consolas</family>
            </font>
           </property>
           <property name="lineWrapMode">
            <enum>QPlainTextEdit::NoWrap</enum>
           </property>
           <property name="readOnly">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QLabel" name="label_16">
           <property name="text">
            <string>Private Key</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QPlainTextEdit" name="CertSignReq">
           <property name="font">
            <font>
             <family>Consolas</family>
            </font>
           </property>
           <property name="lineWrapMode">
            <enum>QPlainTextEdit::NoWrap</enum>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Certificate</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_18">
           <property name="text">
            <string>Certificate Signing Request</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1401</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionSave"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>btnCreateRoot</tabstop>
  <tabstop>btnCreateSub</tabstop>
  <tabstop>btnManage</tabstop>
  <tabstop>leName</tabstop>
  <tabstop>leDescription</tabstop>
  <tabstop>leUID</tabstop>
  <tabstop>leCommonName</tabstop>
  <tabstop>leOrgUnit</tabstop>
  <tabstop>leOrganization</tabstop>
  <tabstop>leLocality</tabstop>
  <tabstop>leStateOrProvince</tabstop>
  <tabstop>leCountry</tabstop>
  <tabstop>leEmail</tabstop>
  <tabstop>leDomain</tabstop>
  <tabstop>lstSubjectAltName</tabstop>
  <tabstop>btnAltNameAdd</tabstop>
  <tabstop>btnAltNameDel</tabstop>
  <tabstop>dtStart</tabstop>
  <tabstop>dtEnd</tabstop>
  <tabstop>rbDigitalSig</tabstop>
  <tabstop>rbKeyEstablishment</tabstop>
  <tabstop>rbNonCompliant</tabstop>
  <tabstop>KeyLength</tabstop>
  <tabstop>CA</tabstop>
  <tabstop>lePathLength</tabstop>
  <tabstop>digitalSignature</tabstop>
  <tabstop>contentCommitment</tabstop>
  <tabstop>keyEncipherment</tabstop>
  <tabstop>dataEncipherment</tabstop>
  <tabstop>keyAgreement</tabstop>
  <tabstop>keyCertSign</tabstop>
  <tabstop>cRLSign</tabstop>
  <tabstop>encipherOnly</tabstop>
  <tabstop>decipherOnly</tabstop>
  <tabstop>btnGenKey</tabstop>
  <tabstop>btnGenCSR</tabstop>
  <tabstop>btnGenCert</tabstop>
  <tabstop>PrivateKey</tabstop>
  <tabstop>CertSignReq</tabstop>
  <tabstop>Certificate</tabstop>
  <tabstop>btnDelete</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
